app/uk/gov/hmrc/fileupload/controllers/EnvelopeChecker.scala

```diff
@@ -20,46 +20,90 @@ import cats.data.Xor
 
-  def withValidEnvelope(check: (EnvelopeId) => Future[EnvelopeStatusResult])
+  val defaultFileSize = 10 * 1024 * 1024
+
+  def withValidEnvelope(checkEnvelopeDetails: (EnvelopeId) => Future[EnvelopeDetailResult])
                        (envelopeId: EnvelopeId)
-                       (action: EssentialAction)
+                       (action: FileSize => EssentialAction)
                        (implicit ec: ExecutionContext) =
     EssentialAction { implicit rh =>
       Accumulator.flatten {
-        check(envelopeId).map {
-          case Xor.Right("OPEN") =>
-            action(rh)
-          case Xor.Right(otherStatus) =>
-            logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $otherStatus")
-          case Xor.Left(EnvelopeStatusNotFoundError(_)) =>
+        checkEnvelopeDetails(envelopeId).map {
+          case Xor.Right(envelope) =>
+            val status = (envelope \ "status").as[String]
+            if (status == "OPEN") {
+              action(setMaxFileSize(envelope))(rh)
+            } else if (status == "CLOSED") {
+              logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $status")
+            } else if(status == "SEALED") {
+              logAndReturn(LOCKED, s"Unable to upload to envelope: $envelopeId with status: $status")
+            } else {
+              //If uploading to deleted Envelope
+              logAndReturn(BAD_REQUEST, s"Unable to upload to envelope: $envelopeId with status: $status")
+            }
+          case Xor.Left(EnvelopeDetailNotFoundError(_)) =>
             logAndReturn(NOT_FOUND, s"Unable to upload to nonexistent envelope: $envelopeId")
           case Xor.Left(error) =>
             logAndReturn(INTERNAL_SERVER_ERROR, error.toString)
         }
       }
     }
 
+  def setMaxFileSize(envelope: JsValue): FileSize = {
+    val definedConstraints = (envelope \ "constraints").asOpt[Constraints]
+       definedConstraints match {
+           //TODO Possibly in the future this may change to take into account the envelope size e.g. if envelope size is 5MB and the upload limit is 10MB
+         case Some(constraints) => constraints.maxSizePerItem match {
+           case Some(maxSizePerItem) =>
+             val fileSize = maxSizePerItem.replaceAll("[^\\d.]", "").toInt
+             val fileSizeType = maxSizePerItem.toUpperCase.replaceAll("[^KMB]", "")
+             fileSizeType match {
+               case "KB" =>
+                 if(fileSize <= 1024) {
+                   fileSize * 1024
+                 } else {
+                   //If maxSizePerItem is accidentally more than 10MB e.g. 10000KB
+                   defaultFileSize
+                 }
+               case "MB" =>
+                 if(fileSize <= 10){
+                   fileSize * 1024 * 1024
+                 } else {
+                   //If maxSizePerItem is accidentally more than 10MB e.g. 100MB
```

Comment:
Personally not a fan of line comments in the code. Shouldn't be needed when the code is readable enough.